### PR TITLE
Auth nav unit tests

### LINKED ISSUE #150: Unit test for authnavguard middleware
**Why**
Increase our test coverage and give us increased confidence in the auth nav guard/middleware

**What**
* [ ] Test all possible types of from routes 
* [ ] Test all possible types of to routes
* [ ] Test all possible authentication types

**Notes**
Sub issue of #101 
Some tests and template/examples in cloud-web-ui

### PR DESCRIPTION
**Pull request by zhen0**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [x] add/update tests (or don't, for reasons explained below)

## Describe this PR
- Adds unit tests for the auth nav guard middleware
- Closes #150

### CODE REVIEW

File: tests/unit/middleware/authNavGuard.spec.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    })
+
+    const next = jest.fn()
+
+    expect(store.getters['auth0/isAuthorized']).toBe(false)
+
+    await authNavGuard({}, {}, next)
+
+    expect(store.getters['auth0/isAuthorized']).toBe(true)
+
```
Comment (znicholasbrown) [inline tests/unit/middleware/authNavGuard.spec.js:122]: Same comment as above re: adding an assertion that `auth0/authorize` is called

File: tests/unit/middleware/authNavGuard.spec.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      store.commit('user/user', { name: 'test_user' })
+    })
+
+    const next = jest.fn()
+    expect(store.getters['user/userIsSet']).toBe(false)
+
+    await authNavGuard({}, {}, next)
+
+    expect(store.getters['user/userIsSet']).toBe(true)
+
```
Comment (znicholasbrown) [inline tests/unit/middleware/authNavGuard.spec.js:143]: Same comment as above re: adding an assertion that `user/getUser` is called

**Review by znicholasbrown — DISMISSED** [review]
Just a few comments but all the same tweak @zhen0

**Comment by zhen0** [comment]
Thanks for the comments @znicholasbrown - I haven't added a check for 'authenticate' or 'authorize' being called because I could not get spying on store.dispatch or store.actions to work.  I've updated the wording of the tests and added comments to reflect what we are testing and also added an issue in #297 to come back and to figure out that spy. 

I've also updated the tenant store unit tests to remove a mockError that was causing a pre-commit hook error for @whimsicallyson - standard git commit should work for now.

**Review by znicholasbrown — APPROVED** [review]
Approved with the new changes 👍

**Pull request merged**